rust/sbp/src/sbp_string.rs

```diff
@@ -37,6 +39,110 @@ where
     }
 }
 
+impl<T: AsRef<[u8]>> SbpString<T, Unterminated> {
+    /// Checked unterminated SbpString builder, verify last byte is non null
+    pub fn unterminated(data: T) -> Result<Self, UnterminatedError> {
+        match data.as_ref().last() {
+            Some(l) if l != &0 => Ok(Self::new(data)),
+            _ => Err(UnterminatedError),
+        }
+    }
+}
```

Comment:
Seems like we can drop the check here. I'd keep the function for consistency though

```suggestion
impl<T: AsRef<[u8]>> SbpString<T, Unterminated> {
    /// Constructs a new unterminated string
    pub fn unterminated(data: T) -> Self {
        SbpString::new(data)
    }
}
```